### ISSUE TITLE
Modify the structure of the model，but the P=0, R=0,mAP=0?

### ISSUE THREAD

**Issue by ZwNSW** [ultralytics/yolov5#1756]
## ❔Question
@glenn-jocher Recently, I tried to change the order of BatchNorm2d(c2) and Hardswish(c2) in the common.py file to try new improvements. However, after I finished training the code, when I tested my model best.pt in the test.py file, there appeared p=0, R=0,Recall=0,mAP=0. I couldn't find out the reason, so I asked you for advice and looked forward to your reply.

`class Conv(nn.Module):
    # Standard convolution
    def __init__(self, c1, c2, k=1, s=1, p=None, g=1, act=True):  # ch_in, ch_out, kernel, stride, padding, groups
        super(Conv, self).__init__()
        self.conv = nn.Conv2d(c1, c2, k, s, autopad(k, p), groups=g, bias=False)
        #self.bn = nn.BatchNorm2d(c2)
        self.act = nn.Hardswish() if act else nn.Identity()
        self.bn = nn.BatchNorm2d(c2)
     
    #def forward(self, x):
        #return self.act(self.bn(self.conv(x)))

    def forward(self, x):
        return self.bn(self.act(self.conv(x)))

    def fuseforward(self, x):
        return self.act(self.conv(x))`
## Additional context
![image](https://user-images.githubusercontent.com/57338462/102888901-9d4f4a80-4494-11eb-96f8-5f9b824f3f8b.png)

**Comment by glenn-jocher** [comment]
Zero Targets, you have no labels in your test data.

Zero labels will produce zero mAP.

**Comment by ZwNSW** [comment]
@glenn-jocher There are labels in my test set. I just changed the CLass Conv(nn.module) in common.py. As shown in the following code, but training can be done, but when I finished training, P = 0, R = 0 and MAP = 0 appeared in the test. I checked my test set, and there was nothing wrong with it. Because of the same test set, training and testing can be realized in yolov5l.
`
class Conv(nn.Module):
    # Standard convolution
    def __init__(self, c1, c2, k=1, s=1, p=None, g=1, act=True):  # ch_in, ch_out, kernel, stride, padding, groups
        super(Conv, self).__init__()
        self.conv = nn.Conv2d(c1, c2, k, s, autopad(k, p), groups=g, bias=False)
        self.act = nn.Hardswish() if act else nn.Identity()
        self.bn = nn.BatchNorm2d(c2)

    def forward(self, x):
        return self.bn(self.act(self.conv(x)))

    def fuseforward(self, x):
        return self.act(self.conv(x))
`

**Comment by glenn-jocher** [comment]
@ZwNSW there are zero labels in your test set, as indicated in your screenshot.

<img width="1423" alt="Screen Shot 2020-12-22 at 5 53 40 PM" src="https://user-images.githubusercontent.com/26833433/102950124-c6fd7700-447e-11eb-9aa9-9ab534f4483c.png">

**Comment by ZwNSW** [comment]
@glenn-jocher There are labels in my test set. I just changed the CLass Conv(nn.module) in common.py. As shown in the following code, but training can be done, but when I finished training, P = 0, R = 0 and MAP = 0 appeared in the test. I checked my test set, and there was nothing wrong with it. Because of the same test set, training and testing can be realized in yolov5l.
![image](https://user-images.githubusercontent.com/57338462/102950100-d0312700-4504-11eb-8e82-45a8234fd9d1.png)

**Comment by ZwNSW** [comment]
@glenn-jocher The only change I made was to change the order of Batch Normalization and Hardswish, and it turned out that I could train but not test. But there is nothing wrong with my test set.

**Comment by glenn-jocher** [comment]
@ZwNSW there are zero labels in your test set, as indicated in your screenshot.

**Comment by ZwNSW** [comment]
@glenn-jocher The target can be displayed now, but R, R and mAP are still 0.

![image](https://user-images.githubusercontent.com/57338462/102966005-f6b58900-4529-11eb-8aab-3d6d7dcbcfb4.png)

**Comment by glenn-jocher** [comment]
@ZwNSW then your trained model is incapable of detecting objects. You should follow the directions in the custom training tutorial, and start with COCO128, verifying your pipeline there before moving to custom data. 
https://docs.ultralytics.com/yolov5/tutorials/train_custom_data

**Comment by G-Apple1** [comment]
> @glenn-jocher The only change I made was to change the order of Batch Normalization and Hardswish, and it turned out that I could train but not test. But there is nothing wrong with my test set.

I have the same problem, have you solved it?
